### ISSUE TITLE
Spread + question sign support

### ISSUE THREAD

**Issue by zxqfox** [Kuniwak/jsdoctypeparser#28]
Hey there. I've got an issue about `{...?}`.

What you think about supporting this?

Actual:
```
Syntax Error
Expected "!", "(", "*", "-", "...", "0b", "0o", "0x", "=", "?", "\"", "external", "function", "module", "{", [ \t\r\n ], [0-9.] or [a-zA-Z_$] but end of input found.
```

Looks like it's not a big deal.

Ref: https://github.com/jscs-dev/node-jscs/issues/1577#issuecomment-133671610

/cc @kobezzza

**Comment by Kuniwak** [comment]
Thanks, this bug fixed on v2.0.0-alpha-3.
Please test it by [online demo](http://kuniwak.github.io/jsdoctypeparser/).

**Comment by zxqfox** [comment]
:+1: 

That was quickly ;-) Thanks!
